api/types/role.go

```diff
@@ -243,6 +243,9 @@ type Role interface {
 	GetGroupLabels(RoleConditionType) Labels
 	// SetGroupLabels sets the map of group labels this role is allowed or denied access to.
 	SetGroupLabels(RoleConditionType, Labels)
+
+	// String provides a safe to log representation of the role
+	String() string
```

Comment:
Please don't add String() methods to the already-bloated resource interfaces. Loggers and the fmt package will use the String() method of the underlying type automatically, so this isn't needed.